crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_spread.rs

```diff
@@ -44,8 +50,57 @@ pub(crate) fn unnecessary_spread(checker: &mut Checker, keys: &[Option<Expr>], v
         if let (None, value) = item {
             // We only care about when the key is None which indicates a spread `**`
             // inside a dict.
-            if let Expr::Dict(_) = value {
-                let diagnostic = Diagnostic::new(UnnecessarySpread, value.range());
+            if let Expr::Dict(dict) = value {
+                let mut diagnostic = Diagnostic::new(UnnecessarySpread, value.range());
+                if checker.settings.preview.is_enabled() {
+                    // Delete the `**{`
+                    let tokenizer = BackwardsTokenizer::up_to(
```

Comment:
Is there any way to restructure this to use `SimpleTokenizer`, i.e., forwards lexing? We've considered removing the backwards lexer in the past since it's quite complex, so I'd prefer to minimize usages if possible.

Reply:
Yeah, I think that should be possible:

* For deleting `**{`, I can lex forward from the end of the previous value in the outer dictionary.
* For deleting the last `}`, I can lex forward from the last value of the inner dictionary.

I might have to change the function signature to passi n the whole dict (and not just the keys + values) to handle the case where `{**{"a": "b"}}` case where there's no previous value (in which case I'd lex forward from the dict start).

Reply:
Ok -- I think this should work.

I ended up leaving the formatting fairly ugly -- the more I tried to make it look nice, the less confident I became that I handled all the edge-cases correctly (e.g. handling comments correctly, avoiding double commas, etc) so I ended up going for something simple and just letting an autoformatter figure it out afterwards.

Reply:
Thanks for doing a second pass here!